src/ILVerification/src/VerificationResult.cs

```diff
@@ -7,18 +7,38 @@
 namespace ILVerify
 {
     public class VerificationResult
-    {
-        public MethodDefinitionHandle Method { get; internal set; }
+    {        
         public VerificationErrorArgs Error { get; internal set; }
         public string Message { get; internal set; }
     }
 
-    public struct VerificationErrorArgs
+    public class MethodVerificationResult : VerificationResult
```

Comment:
It may be useful to submit this refactoring in a separate PR to make faster more incremental progress on this. Just do the refactoring that adds ability to do type verification, without actually adding any interesting type verification rules (e.g. just add one very trivial rule - e.g. if the type has no parent, it has to be `System.Object` in core assembly).

Reply:
@jkotas I did this refactor because I would print token values like PEverify and all ILVerifier methods return instance of VerificationResult.Do you mean add this new properties directly to current VerificationResult togheter with method ones and load only what we need related to check type?

Reply:
This refactoring sounds reasonable in general. I am not sure about some of the details - it is not clear why we need a dual type hierarchy for both `VerificationResult` and `VerificationErrorArgs`.

This change has two major parts: (1) Adds plumbing for type verification, (2) Add virtual method implementation verification. Each of these changes is non-trivial. My suggestion was to split this into two PRs - first PR for (1) and second PR for (2).

Reply:
ok, if you agree I'll close this PR and open other one with part 1.

Reply:
I agree